google/cloud/storage/examples/storage_bucket_samples.cc

```diff
@@ -435,6 +435,81 @@ void GetUniformBucketLevelAccess(google::cloud::storage::Client client,
   (std::move(client), argv.at(0));
 }
 
+void SetPublicAccessPreventionEnforced(google::cloud::storage::Client client,
+                                       std::vector<std::string> const& argv) {
+  // [START storage_set_public_access_prevention_enforced]
+  namespace gcs = google::cloud::storage;
+  using google::cloud::StatusOr;
+  [](gcs::Client client, std::string const& bucket_name) {
+    gcs::BucketIamConfiguration configuration;
+    configuration.public_access_prevention = "enforced";
+    StatusOr<gcs::BucketMetadata> updated_metadata = client.PatchBucket(
+        bucket_name, gcs::BucketMetadataPatchBuilder().SetIamConfiguration(
+                         std::move(configuration)));
+
+    if (!updated_metadata) {
+      throw std::runtime_error(updated_metadata.status().message());
+    }
+
+    std::cout << "Public Access Prevention is set to 'enforced' for "
+              << updated_metadata->name() << "\n";
+  }
+  // [END storage_set_public_access_prevention_enforced]
+  (std::move(client), argv.at(0));
+}
+
+void SetPublicAccessPreventionUnspecified(
+    google::cloud::storage::Client client,
+    std::vector<std::string> const& argv) {
+  // [START storage_set_public_access_prevention_unspecified]
+  namespace gcs = google::cloud::storage;
+  using google::cloud::StatusOr;
+  [](gcs::Client client, std::string const& bucket_name) {
+    gcs::BucketIamConfiguration configuration;
+    configuration.public_access_prevention = "unspecified";
+    StatusOr<gcs::BucketMetadata> updated_metadata = client.PatchBucket(
+        bucket_name, gcs::BucketMetadataPatchBuilder().SetIamConfiguration(
+                         std::move(configuration)));
+
+    if (!updated_metadata) {
+      throw std::runtime_error(updated_metadata.status().message());
+    }
+
+    std::cout << "Public Access Prevention is set to 'unspecified' for "
+              << updated_metadata->name() << "\n";
+  }
+  // [END storage_set_public_access_prevention_unspecified]
+  (std::move(client), argv.at(0));
+}
+
+void GetPublicAccessPrevention(google::cloud::storage::Client client,
+                               std::vector<std::string> const& argv) {
+  // [START storage_get_public_access_prevention]
+  namespace gcs = google::cloud::storage;
+  using google::cloud::StatusOr;
+  [](gcs::Client client, std::string const& bucket_name) {
+    StatusOr<gcs::BucketMetadata> bucket_metadata =
+        client.GetBucketMetadata(bucket_name);
+    if (!bucket_metadata) {
+      throw std::runtime_error(bucket_metadata.status().message());
+    }
+
+    if (bucket_metadata->has_iam_configuration() &&
+        bucket_metadata->iam_configuration()
+            .public_access_prevention.has_value()) {
```

Comment:
My understanding is that, after this change, the IAM config and the PAP field should always be present for a bucket. So I don't think this check should be necessary (unless I'm missing something about how things work in this library). @shaffeeullah correct me if I'm wrong.

Reply:
This is defensive, yes.  Having said that: even for all the old buckets?  I doubt the millions (at a guess, but maybe billions) of existing buckets will get the new field added to their metadata.  Even if they do, the application can use a `fields` attribute to selectively return just a few of the fields in the metadata, I would like the code not to crash in that case.